pipeline.py

```diff
@@ -115,35 +114,71 @@ def pipeline_wrapper(mock: List[Literal[MOCKED_STAGES]]):
         description="InstructLab pipeline",
     )
     def pipeline(
-        num_instructions_to_generate: int = 2,
-        repo_url: str = "https://github.com/instructlab/taxonomy.git",
-        repo_branch: Optional[str] = None,
-        repo_pr: Optional[int] = None,
-        storage_class_name: str = "nfs-csi",
-        base_model: str = BASE_MODEL,
-        # minimal subset of MMLU_TASKS
-        # mmlu_tasks_list: str = MMLU_TASKS_LIST,
-        model_dtype: str = MODEL_DTYPE,
-        few_shots: int = FEW_SHOTS,
-        batch_size: str = BATCH_SIZE,
-        max_workers: str = MAX_WORKERS,
-        merge_system_user_message: bool = MERGE_SYSTEM_USER_MESSAGE,
-        device: str = None,
-        nproc_per_node: int = 3,
-        nnodes: int = 2,
-        num_epochs_phase_1: int = NUM_EPOCHS_PHASE_1,
-        num_epochs_phase_2: int = NUM_EPOCHS_PHASE_2,
-        effective_batch_size_phase_1: int = EFFECTIVE_BATCH_SIZE_PHASE_1,
-        effective_batch_size_phase_2: int = EFFECTIVE_BATCH_SIZE_PHASE_2,
-        learning_rate_phase_1: float = LEARNING_RATE_PHASE_1,
-        learning_rate_phase_2: float = LEARNING_RATE_PHASE_2,
-        num_warmup_steps_phase_1: int = NUM_WARMUP_STEPS_PHASE_1,
-        num_warmup_steps_phase_2: int = NUM_WARMUP_STEPS_PHASE_2,
-        save_samples: int = SAVE_SAMPLES,
+        # SDG phase
+        sdg_repo_url: str = "https://github.com/instructlab/taxonomy.git",
+        sdg_repo_branch: Optional[str] = None,
+        sdg_repo_pr: Optional[int] = None,
+        sdg_base_model: str = BASE_MODEL,
+        sdg_scale_factor: int = 2,  # Renamed upstream https://github.com/instructlab/instructlab/blob/f7d40f6ed5112d59132dd832bd332fa6fbbe7010/src/instructlab/configuration.py#L279-L290
         sdg_pipeline: str = SDG_PIPELINE,
-        max_batch_len: int = MAX_BATCH_LEN,
-        seed: int = SEED,
+        sdg_max_batch_len: int = MAX_BATCH_LEN,
+        # Training phase
+        train_nproc_per_node: int = 3,
+        train_nnodes: int = 2,
+        train_num_epochs_phase_1: int = NUM_EPOCHS_PHASE_1,
+        train_num_epochs_phase_2: int = NUM_EPOCHS_PHASE_2,
+        train_effective_batch_size_phase_1: int = EFFECTIVE_BATCH_SIZE_PHASE_1,
+        train_effective_batch_size_phase_2: int = EFFECTIVE_BATCH_SIZE_PHASE_2,
+        train_learning_rate_phase_1: float = LEARNING_RATE_PHASE_1,
+        train_learning_rate_phase_2: float = LEARNING_RATE_PHASE_2,
+        train_num_warmup_steps_phase_1: int = NUM_WARMUP_STEPS_PHASE_1,
+        train_num_warmup_steps_phase_2: int = NUM_WARMUP_STEPS_PHASE_2,
+        train_save_samples: int = SAVE_SAMPLES,
+        train_max_batch_len: int = MAX_BATCH_LEN,
+        train_seed: int = SEED,
+        # MT Bench
+        mt_bench_max_workers: str = MAX_WORKERS,
+        # Final evaluation
+        eval_few_shots: int = FEW_SHOTS,
+        eval_batch_size: str = BATCH_SIZE,
```

Comment:
should these both be `final_eval_` just to keep naming consistency with the step name?

Reply:
I have no preference... does it convey any message to the end user? After all, they will get this context from the description field anyways...
![image](https://github.com/user-attachments/assets/c66b1313-3ff4-487e-ad9d-48f29222c543)